query-engine/driver-adapters/src/proxy.rs

```diff
@@ -279,10 +279,14 @@ fn js_value_to_quaint(
             serde_json::Value::Null => QuaintValue::DateTime(None),
             mismatch => panic!("Expected a string in column {}, found {}", column_name, mismatch),
         },
-        ColumnType::Json => match json_value {
-            serde_json::Value::Null => QuaintValue::Json(None),
-            json => QuaintValue::json(json),
-        },
+        ColumnType::Json => {
+            match json_value {
+                serde_json::Value::Null => QuaintValue::Json(None),
+                // JsonNull
+                serde_json::Value::String(s) if s == "null" => QuaintValue::Text(Some(s.into())),
```

Comment:
Technically storing a string equal to `"null"` is legal and should be possible in JSON columns (although unlikely in practice), should we at least make this `"__prisma_null"` or something like that?

Reply:
Currently that's the specific definition of JsonNull according to tests. A string stored equal to "null" in a JSON column would be "\"null"\" in JSON, am I right?

Reply:
Could you come up with a test that breaks for driver adapters and passes for rust adapters, exercising the scenario you discuss?

Reply:
> A string stored equal to `"null"` in a JSON column would be `"\"null\""` in JSON, am I right?

Yes. But this PR also makes `"\"null\""` the intermediate representation of `null` on the JS -> Rust boundary in driver adapters (to work around the fact the normal `"null"` representation will be parsed to `serde_json::Value::Null` that we treat as DbNull here).

My comment only refers to this intermediate representation, not to the way we store JSON nulls (there's no problem there).

> Could you come up with a test that breaks for driver adapters and passes for rust adapters, exercising the scenario you discuss?

Sure: 591d1446671b8b26652192e1a0b7d638aac09149

Reply:
Since the problem is only about reading and not writing data to DB, it's not likely in practice, and my suggestion doesn't fundamentally solve the problem but just suggests using another, less common string, let's just merge this for now and continue discussion tomorrow. I'll open a new issue for the edge case I'm describing.

Reply:
Ok, I get it now. Please go ahead and add an intermediate string that we can use in the boundaries.